diff --git a/include/flatbuffers/util.h b/include/flatbuffers/util.h
index 1cd4e8f..e0392a9 100644
--- a/include/flatbuffers/util.h
+++ b/include/flatbuffers/util.h
@@ -101,7 +101,7 @@ template<typename T> size_t IntToDigitCount(T t) {
   // Count a single 0 left of the dot for fractional numbers
   if (-1 < t && t < 1) digit_count++;
   // Count digits until fractional part
-  T eps = std::numeric_limits<float>::epsilon();
+  T eps = std::numeric_limits<T>::epsilon();
   while (t <= (-1 + eps) || (1 - eps) <= t) {
     t /= 10;
     digit_count++;
